com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/DefaultRegistry.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Runtime.Serialization;
+using System.Collections.Generic;
+using UnityEditor.ShaderGraph.GraphDelta;
+using System.Linq;
+using UnityEditor.ShaderGraph.Registry.Defs;
+using UnityEditor.ShaderGraph.Registry.Types;
+using com.unity.shadergraph.defs;
+
+namespace UnityEditor.ShaderGraph.Registry.Default
+{
+    public class DefaultContext : IContextDescriptor
+    {
+        public IReadOnlyCollection<IContextDescriptor.ContextEntry> GetEntries()
+        {
+            return new List<IContextDescriptor.ContextEntry>()
+            {
+                new ()
+                {
+                    fieldName = "BaseColor",
+                    primitive = GraphType.Primitive.Float,
+                    precision = GraphType.Precision.Fixed,
+                    height = GraphType.Height.One,
+                    length = GraphType.Length.One,
+                }
+            };
+        }
+
+        public RegistryFlags GetRegistryFlags()
+        {
+            return RegistryFlags.Base;
+        }
+
+        public RegistryKey GetRegistryKey()
+        {
+            // Defines the name of the context node on the graph
+            return new RegistryKey() { Name = "DefaultContextDescriptor", Version = 1 };
+        }
+    }
+
+    public static class DefaultRegistry
+    {
+        public static Registry CreateDefaultRegistry()
+        {
+            var reg = new Registry();
+            reg.Register<Types.GraphType>();
+            reg.Register<Types.GraphTypeAssignment>();
+            reg.Register<DefaultContext>();
+            //RegistryInstance.Register<Registry.Types.AddNode>();
+
+            // Register nodes from FunctionDescriptors in IStandardNode classes.
+            var interfaceType = typeof(IStandardNode);
+            var types = AppDomain.CurrentDomain.GetAssemblies()
+                .SelectMany(s => s.GetTypes())
+                .Where(p => interfaceType.IsAssignableFrom(p));
+            string m = "get_FunctionDescriptor";
+            foreach (var t in types)
+            {
+                var fdMethod = t.GetMethod(m);
+                if (t != interfaceType && fdMethod != null)
+                {
+                    var fd = (FunctionDescriptor)fdMethod.Invoke(null, null);
+                    reg.Register(fd);
+                }
+            }
+            return reg;
```

Comment:
@brett-groshong Just because we moved this from the Stencil, but it sounds like it will up the cost of fetching ui strings and other such if it's wrapped in this way.

We may want a generalized static runtime object that can lazy initialize this stuff (and cache strings and UI element hints) for the duration of the project... need to think on it more.